Home Landing Page UI implementation of Structural Biology Platform

vtnphan opened the pull request:
##  SBP-165: Implement HOME Landing Page UI

### Summary
[SBP-165](https://biocloud.atlassian.net/browse/SBP-165)
This PR implements the complete HOME landing page UI following the [Adobe XD design specifications](https://xd.adobe.com/view/21a2a6b2-16c2-48ff-afc2-47044ddd7798-74f1/). The implementation includes reusable header and footer components, with proper theme and workflow state management as outlined in the acceptance criteria.

###  Acceptance Criteria Met
- [x] **Page displays designed footer as reusable component** - Implemented responsive footer with links, contact info, and proper styling
- [x] **Page displays designed header as reusable component** - Created header component with navigation, authentication, and tab management
- [x] **Theme: Binder design active (other tabs inactive)** - Default active tab set to "binder-design" with proper state management
- [x] **Workflow: Hotspot-guided binder design pill active** - Implemented workflow selection with "hotspot-guided binder design" as default active
- [x] **Tool: BindCraft (active), RFdiffusion (inactive)** - Tool selection implemented with BindCraft as default active tool

### Key Features Implemented

#### Navigation & Layout
- **Header Component**: Responsive navigation with theme tabs, search functionality, and authentication integration
- **Footer Component**: Multi-section footer with links, contact information, and social media
- **Navbar Component**: Mobile-responsive navigation with dropdown menus and active state management

#### Pages & Components
- **Home Page**: Main landing page implementing the designed UI
- **Binder Design Page**: Specialized page for binder design workflows
- **Not Found Page**: Custom 404 error page
- **Loading Component**: Global loading indicator

#### Styling & UI
- **SCSS Variables**: Centralized design system with consistent colors and spacing
- **Component Styling**: Modular SCSS files for each component

### Files Changed
- **71 files changed**: +5,202 additions, -565 deletions
- **New Components**: 10+ reusable Angular components
- **Enhanced Services**: Improved authentication and error handling
- **Test Coverage**: Comprehensive test suite with high coverage

### Demo 
[Demo link](https://biocloud.atlassian.net/browse/SBP-165?focusedCommentId=20484)
This PR successfully delivers a production-ready HOME landing page that meets all design specifications and acceptance criteria, with comprehensive testing and maintainable code architecture.

[SBP-165]: https://biocloud.atlassian.net/browse/SBP-165?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

marius-mather commented on src/app/components/footer/footer.component.html:
```diff
@@ -0,0 +1,163 @@
+        <div class="lg:col-span-3 sm:col-span-2 text-center">
+          <div class="grid grid-cols-2">
+            <a
+              *ngFor="let logo of logos.slice(0, 2)"
```
Why does this need to be done with separate handling for the first 2 logos? Can it be done with a single loop?

vtnphan replied:
Yeah, tbh i cant think of any other way T_T. It is said we should have the same (or similar) footer from Biocommons website. There're 3 logos there, the first two are placed side by side in the first row and left-aligned, while the third logo appears below them in the second row, also left-aligned. The layout remains consistent, regardless of window resizing or screen width changes (I tested).

marius-mather replied:
OK no worries - if we need a specific layout it's fine to write extra code to achieve it

marius-mather commented on src/app/components/footer/footer.component.html:
```diff
@@ -0,0 +1,163 @@
+        <div class="lg:col-span-3 sm:col-span-2 text-center">
+          <div class="grid grid-cols-2">
+            <a
+              *ngFor="let logo of logos.slice(0, 2)"
```
`ngIf` and `ngFor` are deprecated, better to use the new control flow syntax: https://dev.to/vetriselvan_11/angular-20-control-flow-part-1-say-goodbye-to-ngif-ngfor-and-ngswitch-42d8

marius-mather commented on src/app/pages/home/home.html:
```diff
@@ -0,0 +1,43 @@
+    </div>
+
+    <!-- Structure Prediction Tab -->
+    <div *ngIf="isActiveTab('structure-prediction')" class="space-y-6">
```
Can you do these tabs (apart from binder design) by looping over the `tabs` variable?

marius-mather commented on src/app/pages/home/home.scss:
```diff
@@ -0,0 +1,219 @@
+}
+
+// Chip styling for Binder Design
+.bg-biocommons-primary {
```
you should be able to add custom colours to your tailwind config and have things like `bg-biocommons-primary` work automatically.

marius-mather commented on src/styles.scss:
```diff
@@ -0,0 +1,176 @@
+// ===================================================
+
+// Primary color utilities
+.text-primary-pink {
```
See https://tailwindcss.com/docs/adding-custom-styles#customizing-your-theme - should be possible to define tailwind colours without having to define them separately for `text-`, `bg-`, etc.

marius-mather reviewed (changes requested):
Looks great overall but try to make sure we're not writing extra code where we don't need it - more code is more to maintain. Try to use loops to create repeated elements where possible, and use tailwind classes instead of manually writing custom CSS

vtnphan commented:
@marius-mather  Thanks for all the review and comments !!
Some modification was updated to simplify the scss and some functions:

- Migrate to the Angular flow control syntax at https://github.com/AustralianBioCommons/sbp-portal/pull/4/commits/9566bc896695995d11a7d1978c64072cec691594 
- Migrate to taiwind from custom scss, and simplify the components' scss at https://github.com/AustralianBioCommons/sbp-portal/pull/4/commits/168962c45d9bcdaf3d797b423fe411586986ea3c
- queryParams and loop function for tabs at https://github.com/AustralianBioCommons/sbp-portal/pull/4/commits/01b6678d8ec4c912694fa78e85f5fd4b73c68433 and https://github.com/AustralianBioCommons/sbp-portal/pull/4/commits/2fac16b89a1c9492ab3480f5fee2acde036e87ce 
Also added the nav bar menu list items according to Thom's cmt [here](https://biocloud.atlassian.net/browse/SBP-165?focusedCommentId=20485)

Updated screenshot is below, highlighted by red rectangles
<img width="1453" height="819" alt="Screenshot 2025-11-04 at 8 50 56 am" src="https://github.com/user-attachments/assets/54f8883c-118c-4b3b-847d-26aac5a13b7b" />

marius-mather commented on src/app/components/header/header.component.scss:
```diff
@@ -0,0 +1,21 @@
+}
+
+// Tab navigation styles
+.text-biocommons-primary {
```
I think you shouldn't need these kinds of definitions, you have the colours defined in your Tailwind config (like this https://v3.tailwindcss.com/docs/customizing-colors#using-custom-colors), so `text-biocommons-primary-pink` should be automatically generated by Tailwind

marius-mather reviewed (changes requested):
Looking good now - I like that we've been able to reduce the amount of custom CSS. just pointed out a couple of places where I think we could remove more custom CSS

marius-mather reviewed (dismissed):
Thanks for all the changes - looking good now

vtnphan commented:
@marius-mather I've just fixed a text color in the latest commit! Can you please help review again, and it will be ready to be merged

marius-mather reviewed (approved):
approved again
